pkg/objstore/oci/helper.go

```diff
@@ -0,0 +1,270 @@
+		TLSClientConfig:       &tls.Config{InsecureSkipVerify: config.HTTPConfig.InsecureSkipVerify},
+		MaxIdleConns:          config.HTTPConfig.MaxIdleConns,
+		MaxIdleConnsPerHost:   config.HTTPConfig.MaxIdleConnsPerHost,
+		MaxConnsPerHost:       config.HTTPConfig.MaxConnsPerHost,
+		DisableCompression:    config.HTTPConfig.DisableCompression,
+	}
+}
+
+func getNamespace(client objectstorage.ObjectStorageClient, requestMetadata common.RequestMetadata) (namespace *string, err error) {
+	response, err := client.GetNamespace(
+		context.Background(),
+		objectstorage.GetNamespaceRequest{RequestMetadata: requestMetadata},
+	)
+	if err != nil {
+		return nil, err
+	}
+	return response.Value, nil
+}
+
+func getObject(ctx context.Context, bkt Bucket, objectName string, byteRange string) (response objectstorage.GetObjectResponse, err error) {
+	if len(objectName) == 0 {
+		err = fmt.Errorf("value cannot be empty for field ObjectName in path")
+		return
+	}
+	request := objectstorage.GetObjectRequest{
+		NamespaceName:   &bkt.namespace,
+		BucketName:      &bkt.name,
+		ObjectName:      &objectName,
+		RequestMetadata: bkt.requestMetadata,
+	}
+	if byteRange != "" {
+		request.Range = &byteRange
+	}
+	return bkt.client.GetObject(ctx, request)
+}
+
+func listAllObjects(ctx context.Context, bkt Bucket, prefix string, options ...objstore.IterOption) (objectNames []string, err error) {
+	var allObjectNames []string
+	var nextStartWith *string = nil
+	init := true
+
+	for init || nextStartWith != nil {
+		init = false
+		objectNames, nextStartWith, err = listObjects(ctx, bkt, prefix, nextStartWith)
+		if err != nil {
+			return nil, err
+		}
+
+		if objstore.ApplyIterOptions(options...).Recursive {
+			for _, objectName := range objectNames {
+				if strings.HasSuffix(objectName, DirDelim) {
+					subObjectNames, err := listAllObjects(ctx, bkt, objectName, options...)
+					if err != nil {
+						return nil, err
+					}
+					allObjectNames = append(allObjectNames, subObjectNames...)
+				} else {
+					allObjectNames = append(allObjectNames, objectName)
+				}
+			}
+		} else {
+			allObjectNames = append(allObjectNames, objectNames...)
+		}
+	}
+	return allObjectNames, nil
+}
+
+func listObjects(ctx context.Context, bkt Bucket, prefix string, start *string) (objectNames []string, nextStartWith *string, err error) {
+	request := objectstorage.ListObjectsRequest{
+		NamespaceName:   &bkt.namespace,
+		BucketName:      &bkt.name,
+		Delimiter:       common.String(DirDelim),
+		Prefix:          &prefix,
+		Start:           start,
+		RequestMetadata: bkt.requestMetadata,
+	}
+	response, err := bkt.client.ListObjects(ctx, request)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for _, object := range response.ListObjects.Objects {
+		objectNames = append(objectNames, *object.Name)
+	}
+	objectNames = append(objectNames, response.ListObjects.Prefixes...)
+
+	return objectNames, response.NextStartWith, nil
+}
+
+func (config *Config) validateConfig() (err error) {
```

Comment:
nitpick; For usability, we can improve this by validating everything before returning. For example https://github.com/thanos-io/thanos/blob/main/pkg/objstore/azure/azure.go#L149

Reply:
Good idea. Just pushed an updated.